sound/soc/codecs/max98373-sdw.c

```diff
@@ -333,17 +331,6 @@ static int max98373_read_prop(struct sdw_slave *slave)
 		i++;
 	}
 
-	/* Allocate port_ready based on num_of_ports */
-	slave->port_ready = devm_kcalloc(&slave->dev, num_of_ports,
-					 sizeof(*slave->port_ready),
-					 GFP_KERNEL);
```

Comment:
am I confused or does this mean, that between the previous and this patch `slave->port_ready` was assigned by two allocations? Maybe we can avoid this somehow... E.g. if we embed the array and are forced to merge these two patches?

Reply:
the allocation was initially done either in the mipi_disco stuff, or in the codec driver if it didn't use the mipi_disco stuff.
The point now is that we remove both alternatives and do a single allocation in a single place

Reply:
fixed now